homeassistant/components/climate/zha.py

```diff
@@ -0,0 +1,251 @@
+                        discovery_info['manufacturer'],
+                        discovery_info['model'])
+
+
+class ClimateST218(zha.Entity, ClimateDevice):
+    """Climage Stelpro ST218."""
+
+    _domain = DOMAIN
+
+    def __init__(self, **kwargs):
+        """Initialize the ZHA Climate."""
+        super().__init__(**kwargs)
+
+        self._target_temperature = None
+        self._current_temperature = None
+        self._current_operation = None
+        self._heating_power = None
+        self.last_request = time.time()
+
+        # Add Stelpro Specific
+        import bellows.types as t
+        # Remove the following lines when this is merged
+        # https://github.com/rcloran/bellows/pull/71
+
+        class enum8(t.uint8_t):  # pylint: disable=C0103
+            """Missing bellows enum8 type."""
+
+            _enum = True
+
+        t.enum8 = enum8
+        from bellows.zigbee.zcl import foundation
+        foundation.DATA_TYPES[0x30] = ('Enumeration', t.enum8,
+                                       foundation.Discrete)
+        foundation.DATA_TYPE_IDX[t.enum8] = 0x30
+        self._endpoint.thermostat.attributes[0x001C] = \
+            ('system_mode', t.enum8)
+        # END
+
+        # pylint: disable=W0212
+        self._endpoint.thermostat._attridx['setpoint_mode_manuf_specific'] = \
+            0x401C
+        # pylint: enable=W0212
+        self._endpoint.thermostat.attributes[0x401C] = \
+            ('setpoint_mode_manuf_specific', t.enum8)
+
+    @asyncio.coroutine
+    def _wait_for_clean_communication(self):
+        """Ensure that we not request the device more than time by second."""
+        if time.time() - self.last_request < CLEAN_TIME_ZONE:
+            yield from asyncio.sleep(CLEAN_TIME_ZONE)
+        self.last_request = time.time()
+
+    @asyncio.coroutine
+    def async_update(self):
+        """Retrieve latest state."""
+        _LOGGER.debug("%s async_update", self.entity_id)
+        yield from self._wait_for_clean_communication()
+        result = yield from safe_read(self._endpoint.thermostat,
+                                      ['local_temp',
+                                       'pi_heating_demand',
+                                       'occupied_heating_setpoint',
+                                       'setpoint_mode_manuf_specific'])
+        self.last_request = time.time()
+        if not result:
+            _LOGGER.warning("No result received")
+            return
+
+        try:
+            self._heating_power = int(result['pi_heating_demand'])
+        except (TypeError, KeyError):
+            _LOGGER.warning("Can not get heating power value")
+            self._heating_power = None
+
+        raw_mode = result.get('setpoint_mode_manuf_specific')
+        if raw_mode not in OPERATIONS:
+            _LOGGER.warning("Mode %s not found in known modes %s",
+                            raw_mode, OPERATIONS)
+        self._current_operation = OPERATIONS[raw_mode]
```

Comment:
This throws if `raw_mode not in OPERATIONS`. Is that intended? I don't think so, as that also causes the temperatures not to be updated.

Reply:
Maybe just missing `else` statement